src/test/java/riemann/java/client/tests/LocalhostResolveTest.java

```diff
@@ -0,0 +1,146 @@
+package riemann.java.client.tests;
+
+import com.aphyr.riemann.client.LocalhostResolver;
+import junit.framework.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.lang.reflect.Field;
+import java.net.UnknownHostException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class LocalhostResolveTest {
+
+    protected Map<String, String> env = new HashMap<String, String>();
+
+    protected static final String OS_NAME_PROP = "os.name";
+    protected static final String ExpectedWinHostname = "LocalHostResolverTestWin";
+    protected static final String ExpectedNixHostname = "LocalHostResolverTestNix";
+
+    @BeforeClass
+    public static void oneTimeSetUp() {
+        LocalhostResolver.RefreshIntervalMillis = 1000;
+    }
+
+    @Before
+    public void setup() {
+        env = new HashMap<String, String>(System.getenv());
+        env.put(LocalhostResolver.HOSTNAME, ExpectedNixHostname);
+        env.put(LocalhostResolver.COMPUTERNAME, ExpectedWinHostname);
+        setEnv(env);
+
+        LocalhostResolver.CustomEnvVarName = null;
+        LocalhostResolver.resetUpdateTimes();
+        Assert.assertEquals(0, LocalhostResolver.getLastUpdateTime());
+        Assert.assertEquals(0, LocalhostResolver.getLastNetUpdateTime());
+    }
+
+    @Test
+    public void testUpdateInterval() {
+        Assert.assertEquals(0, LocalhostResolver.getLastUpdateTime());
+        String hostname = LocalhostResolver.getResolvedHostname();
+        long lastUpdateTime = LocalhostResolver.getLastUpdateTime();
+        Assert.assertNotNull(hostname);
+        try {
+            Thread.sleep(1500);
```

Comment:
you shouldn't use sleep in unit tests. You can expose get/setLastNetUpdateTime() for testing.
You can have a unit test to check that it actually changed the first time called, but did not change the second time called. Then inject a bogus lastNetUpdateTime call again and check that it was updated again.